packages/dds/tree/src/test/utils.ts

```diff
@@ -407,12 +411,20 @@ export class TestTreeProviderLite {
 				id: "test",
 				idCompressor: createIdCompressor(sessionId),
 			});
-			const tree = this.factory.create(runtime, TestTreeProviderLite.treeId) as SharedTree;
-			this.runtimeFactory.createContainerRuntime(runtime);
+			const tree = this.factory.create(
+				runtime,
+				TestTreeProviderLite.treeId,
+			) as SharedTreeWithConnectionStateSetter;
```

Comment:
You can do the `as Mutable<SharedTreeWithConnectionStateSetter>` here. This function is responsible for creating this tree object so it's fine for it to have mutability for its full scope.